src/Analyser/NodeScopeResolver.php

```diff
@@ -2603,7 +2604,8 @@ static function (Node $node, Scope $scope) use ($nodeCallback): void {
 					$hasYield = $hasYield || $armResult->hasYield();
 					$throwPoints = array_merge($throwPoints, $armResult->getThrowPoints());
 					$scope = $scope->mergeWith($matchScope);
-					$armNodes[] = new MatchExpressionArm([], $arm->getLine());
+					$matchArmBody = new MatchExpressionArmBody($scope, $arm->body);
```

Comment:
I think the right scope here is `$matchScope`. But the one that's correct at the beginning of the foreach, before it's rewritten with `$matchScope = $armResult->getScope();`.